### PR TITLE
[notifications] setting up case specific notifications

### PR DESCRIPTION
**Pull request by stephaniewong2**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Set up push notifications for all users involved in the case if they have enabled push notifications. All users should be prompted with an "Allow Notifications" modal on the Cases screen if setting notification settings for the first time. Clicking allow/yes adds the user's `expo_push_token` to the public `users` table which is then used to send notifications to the specific user. The edge function `push` currently sends a notification to the expo push token sent to it. 

Whenever a new row is added to the `updates` table, the database function `send_notification` is triggered which grabs all `expo_push_token`s of involved users and calls the edge function `push` for each user which sends a notification.

You do not need Docker installed unless you want to deploy a new version of the edge function. I believe you will need to install the Deno extension on VSCode.

Add `EXPO_PUBLIC_PROJECT_ID` to the `.env` file.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Make sure user is prompted with Allow Notifications on the `AllCases` tab
- Add a new row to the `updates` table for a case you are involved in. You should get a notification with the `updateInfo` as the body
- Add a new row to the `updates` table for a case you are not involved in. You should not get a notification. 
- Add a new row to the `updates` table for a case you and another user with push notifications enabled are involved in. Both users should get a notification. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Deep linking so that clicking notification leads to updates tab (?)
- Populate Updates screen with updates.
- Configure notifications to also support notifications to all users and new users as mentioned in Alex's doc on Slack. :,(
- Test on Android...

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
[Article on how to set up edge function for emails](https://medium.com/@DeolaJ/how-to-send-emails-with-edge-functions-on-supabase-3653389c07c4) 
[Follow up article on database function and trigger](https://medium.com/@DeolaJ/how-to-trigger-email-notifications-on-a-contact-form-with-supabase-210cf245e57e)
[Deno Documentation For Processing Edge Function Requests](https://docs.deno.com/runtime/manual/)

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ronniebeggs

### CODE REVIEW

File: src/Components/Push.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Notifications.setNotificationHandler({
+  handleNotification: async () => ({
+    shouldShowAlert: true,
+    shouldPlaySound: false,
+    shouldSetBadge: false,
+  }),
+});
+
+async function getDummyUpdate() {
```
Comment (ronniebeggs) [inline src/Components/Push.tsx]: Can we keep notifications config in this file, but eventually move actual notification functions (including example notifications) to a separate `notifications.ts` file in the supabase queries folder. Or somewhere else, idk what necessarily works best (?)

File: src/Components/Push.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        setNotification(notification);
+      });
+
+    responseListener.current =
+      Notifications.addNotificationResponseReceivedListener(response => {
+        console.log(response);
+      });
+
+    return () => {
+      Notifications.removeNotificationSubscription(
```
Comment (ronniebeggs) [inline src/Components/Push.tsx]: what is being returned from this useEffect?

File: src/app/(BottomTabNavigation)/Updates/index.tsx [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+
+    supabase.auth.onAuthStateChange((_event, session) => {
+      setSession(session);
+    });
+  }, []);
+
   return (
     <View style={styles.container}>
       <Text>This is the /Updates/index update screen!</Text>
+      <Push session={session} />
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/Updates/index.tsx]: Typescript needs to ensure that the session is not null.

File: supabase/functions/pushNotification/index.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.38.4';
```
Comment (ronniebeggs) [inline supabase/functions/pushNotification/index.ts]: What's the separate supabase folder for?

**Review by ronniebeggs — COMMENTED** [review]
Admittedly I did not get notifications to work on my end, nor did I entirely understand what everything meant. Although it seems promising. I only left comments on file/function organization stuff that we don't necessarily need to care about, until we try to make everything robust. Looks good so far!

File: src/Components/Push.tsx (hunk 2 above)
Comment (stephaniewong2) [reply src/Components/Push.tsx]: not sure... removed it bc i just copy and pasted from the example originally

**Comment by ronniebeggs** [comment]
@stephaniewong2 Waiting on Noah's response on how to resolve typescript issues. Otherwise it looks good to merge!
